Add Matrix tests and fix compat bugs

hughbe opened the pull request:
Related: https://github.com/dotnet/corefx/pull/24238

dnfclas commented:
@hughbe,
Thanks for having already signed the Contribution License Agreement. Your agreement was validated by .NET Foundation. We will now review your pull request.
_Thanks,
.NET Foundation Pull Request Bot_

hughbe commented on tests/testmatrix.c:
```diff
@@ -0,0 +1,799 @@
+	assert (status == InvalidParameter);
+
+	// This causes a null pointer dereference in GDI+.
+#if !defined(USE_WINDOWS_LIBGDIPLUS)
```
Another GDI+ bug that I don't think is worth replicating

hughbe commented on tests/testmatrix.c:
```diff
@@ -679,6 +679,7 @@ static void test_isMatrixIdentity ()
 	assert (isIdentity == TRUE);
 
 	// Less close to identity.
+	// FIXME: it appears that GDI+ is less precise than GDI+.
```
@mellinoe @akoeplinger do you know the algorithm GDI+ uses for `GdipIsMatrixIdentity`? I've tried narrowing the values down to around `+-0.99949f` but I don't want to start mucking around with things that potentially break

mellinoe replied:
Sorry, I don't know -- I'm really not familiar with the GDI+ codebase.

hughbe replied:
No problem, just checking!

akoeplinger commented on tests/testmatrix.c:
```diff
@@ -0,0 +1,799 @@
+		return fabs (v1 - v2) < 0.0001;
+}
+
+void verifyMatrix (GpMatrix *matrix, REAL e1, REAL e2, REAL e3, REAL e4, REAL e5, REAL e6)
```
this and floatsEqual already exist in testhelpers.h, please just #include that and use it from there (and while you're at it please get rid of the copy in testpathgradientbrush.c as well).

akoeplinger commented on tests/Makefile.am:
```diff
@@ -110,12 +116,13 @@ testpng_SOURCES =		\
 	$(testgdi_SOURCES)	\
 	$(testbits_SOURCES)	\
-	$(testbrush_SOURCES)	\
+	$(testbrush_SOURCES) \
```
nit: revert the whitespace changes here and below.

hughbe commented:
Good point - I usually copy-paste the code into rextester.com to test the GDI+ code against Windows (then also on my local machine at the end to make sure). Rextester.com doesn't allow including files so I just copy-pasted them and forgot to revert that.

I've addressed the PR feedback, thanks for reviewing!
